ui/src/components/create_client/CreateMobileClientBaseClass.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import { Grid, Form } from 'patternfly-react';
+import { CREATE_CLIENT_APP_ID, CREATE_CLIENT_NAME } from './Constants';
+import { VerticalFormField } from './VerticalFormField';
+
+/**
+ * Component for the Android specific create mobile client form.
+ */
+class CreateMobileClientBaseClass extends Component {
+  constructor(platformName, props) {
+    super(props);
+    this.state = {
+      valid: false,
+      validationState: {}
+    }
+    this.config = {
+      platform: platformName,
+      appName: {
+        label: '* App Name',
+        example: 'myapp',
+        help: 'App name must match ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
```

Comment:
while this is the rule we are using to validate the value, I don't think we should present it in the help message. I think a better help message should be like "app name should only contain letters, numbers and dots".